crates/hotshot/src/lib.rs

```diff
@@ -288,22 +292,25 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
         broadcast_event(event, &self.inner.output_event_stream.0).await;
     }
 
-    /// Publishes a transaction asynchronously to the network
+    /// Publishes an initial transaction asynchronously to the network.
+    ///
+    /// The transaction corresponds to either the genesis view or the view to restart on.
     ///
     /// # Errors
     ///
     /// Always returns Ok; does not return an error if the transaction couldn't be published to the network
     #[instrument(skip(self), err)]
-    pub async fn publish_transaction_async(
+    pub async fn publish_initial_transaction_async(
```

Comment:
I think adding initial to the name makes it feel like this is something you should only call once for a first transaction.  It's still the main transaction submission path.  I think what we should do is fix `consensus.curr_view`.  We can updated it in the consensus task and read it instead of `start_view` here.

I actually noticed during some libp2p stuff that transactions gossips get rejected in tests because they are duplicate messages.  It's because the test transaction payload is always the same, and now seeing that view 0 was hard coded, the view is clearly also always the same.  Would be nice if it updated roughly with consensus to  avoid this issue.

Reply:
The transaction submission functions are currently only used for the first view, so I thought it was okay to add `initial` to the names. But I do think it's a better idea to fix `cur_view` instead. Will do that!